src/BlockEntities/EnderChestEntity.cpp

```diff
@@ -48,8 +49,16 @@ void cEnderChestEntity::SendTo(cClientHandle & a_Client)
 
 bool cEnderChestEntity::UsedBy(cPlayer * a_Player)
 {
-	// TODO: cats are an obstruction
-	if ((GetPosY() < cChunkDef::Height - 1) && !cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ())))
+	if (
+		(GetPosY() < cChunkDef::Height - 1) &&
+		(
+			!cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ())) ||
+			(
+				(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ()) == E_BLOCK_AIR) &&
+				!cOcelot::IsCatSittingOnBlock(GetWorld(), Vector3d(GetPos()))
```

Comment:
Does this mean that a cat sitting on a chest that has a torch above it still doesn't block the chest from opening? Is that vanilla behavior?